client.go

```diff
@@ -519,109 +513,118 @@ func WithImportLabels(labels map[string]string) ImportOpt {
 	}
 }
 
-// WithOCIImportFormat sets the import format for an OCI image format
-func WithOCIImportFormat() ImportOpt {
+// WithImporter sets the importer
+func WithImporter(importer imageformats.Importer) ImportOpt {
 	return func(c *importOpts) error {
-		if c.format != "" {
-			return errors.New("format already set")
+		if c.importer != nil {
+			return errors.New("impoter already set")
 		}
-		c.format = ociImageFormat
-		return nil
-	}
-}
-
-// WithRefObject specifies the ref object to import.
-// If refObject is empty, it is copied from the ref argument of Import().
-func WithRefObject(refObject string) ImportOpt {
-	return func(c *importOpts) error {
-		c.refObject = refObject
+		c.importer = importer
 		return nil
 	}
 }
 
-func resolveImportOpt(ref string, opts ...ImportOpt) (importOpts, error) {
+func resolveImportOpt(opts ...ImportOpt) (importOpts, error) {
 	var iopts importOpts
 	for _, o := range opts {
 		if err := o(&iopts); err != nil {
 			return iopts, err
 		}
 	}
 	// use OCI as the default format
-	if iopts.format == "" {
-		iopts.format = ociImageFormat
-	}
-	// if refObject is not explicitly specified, use the one specified in ref
-	if iopts.refObject == "" {
-		refSpec, err := reference.Parse(ref)
-		if err != nil {
-			return iopts, err
-		}
-		iopts.refObject = refSpec.Object
+	if iopts.importer == nil {
+		iopts.importer = oci.V1Importer
 	}
 	return iopts, nil
 }
 
 // Import imports an image from a Tar stream using reader.
 // OCI format is assumed by default.
+// Filter strings are format-specific, but needs to match exactly one item.
+//
+// TODO(AkihiroSuda): support importing multiple images from a single archive at once? (map[ref]filter?)
```

Comment:
RFC

Reply:
Yes, I think this is a must.

Reply:
For us, we only need loading one image from tarball for now. We only build one image into each tar in the Kubernetes building process.